tests/unit/phpDocumentor/Application/Commands/ConfigureCacheHandlerTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Application\Commands;
+
+use Mockery as m;
+use org\bovigo\vfs\vfsStream;
+use phpDocumentor\Application\Configuration\ConfigurationFactory;
+use phpDocumentor\Application\Configuration\Factory\Strategy;
+use phpDocumentor\Uri;
+use PHPUnit_Framework_TestCase;
+use Stash\Pool;
+
+/**
+ * @coversDefaultClass phpDocumentor\Application\Commands\ConfigureCacheHandler
+ * @covers ::__construct
+ */
+final class ConfigureCacheHandlerTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var m\MockInterface|Pool
+     */
+    private $pool;
+
+    /**
+     * @var ConfigurationFactory
+     */
+    private $configurationFactory;
+
+    /**
+     * @var ConfigureCacheHandler
+     */
+    private $fixture;
+
+    /**
+     * @var m\MockInterface
+     */
+    private $configStrategy;
+
+    /**
+     *
+     */
+    protected function setUp()
+    {
+        $this->pool = m::mock(Pool::class);
+        $this->configStrategy = m::mock(Strategy::class);
+        $this->configStrategy->shouldReceive('match')->andReturn(true);
+
+        $root = vfsStream::setup('dir');
+
+        vfsStream::newFile('foo.xml')->at($root)->withContent('<foo></foo>');
+        $uri = new Uri(vfsStream::url('dir/foo.xml'));
+
+        $this->configurationFactory = new ConfigurationFactory([$this->configStrategy], $uri);
+        $this->fixture = new ConfigureCacheHandler($this->pool, $this->configurationFactory);
+    }
+
+    /**
+     * @covers ::__invoke
+     */
+    public function testDisableCache()
+    {
+        $this->invokeTest(false);
+    }
+
+    /**
+     * @covers ::__invoke
+     */
+    public function testLeaveCache()
+    {
+        $this->invokeTest(true);
+    }
+
+    private function invokeTest($useCache)
```

Comment:
Wouldn't it be better to make this the test and have a data provider insert the true and false?